src/EditorFeatures/TestUtilities/QuickInfo/ToolTipAssert.cs

```diff
@@ -0,0 +1,241 @@
+                for (var i = 0; i < runs.Length; i++)
+                {
+                    ContainerToString(runs[i], indent, result);
+
+                    if (i < runs.Length - 1)
+                        result.AppendLine(",");
+                    else
+                        result.Append(")");
+                }
+
+                return;
+            }
+
+            if (element is ClassifiedTextRun classifiedTextRun)
+            {
+                var classification = GetKnownClassification(classifiedTextRun.ClassificationTypeName);
+                result.Append($"{classification}, \"{classifiedTextRun.Text.Replace("\"", "\"\"")}\"");
+                if (classifiedTextRun.NavigationAction is object || !string.IsNullOrEmpty(classifiedTextRun.Tooltip))
+                {
+                    var tooltip = classifiedTextRun.Tooltip is object ? $"\"{classifiedTextRun.Tooltip.Replace("\"", "\"\"")}\"" : "Nothing";
+                    result.Append($", navigationAction:=Sub() Return, {tooltip}");
+                }
+
+                if (classifiedTextRun.Style != ClassifiedTextRunStyle.Plain)
+                {
+                    result.Append($", {TextRunStyleToString(classifiedTextRun.Style)}");
+                }
+
+                result.Append(")");
+                return;
+            }
+
+            throw ExceptionUtilities.Unreachable;
+        }
+
+        private static string ContainerStyleToString(ContainerElementStyle style)
+        {
+            var stringValue = style.ToString();
+            return string.Join(" Or ", stringValue.Split(new[] { ',', ' ' }, StringSplitOptions.RemoveEmptyEntries).Select(value => $"{nameof(ContainerElementStyle)}.{value}"));
+        }
+
+        private static string TextRunStyleToString(ClassifiedTextRunStyle style)
+        {
+            var stringValue = style.ToString();
+            return string.Join(" Or ", stringValue.Split(new[] { ',', ' ' }, StringSplitOptions.RemoveEmptyEntries).Select(value => $"{nameof(ClassifiedTextRunStyle)}.{value}"));
+        }
+
+        private static string GetKnownClassification(string classification)
+        {
+            foreach (var field in typeof(ClassificationTypeNames).GetFields())
+            {
+                if (!field.IsStatic)
+                    continue;
+
+                var rawValue = field.GetValue(null);
+                var value = rawValue as string;
+                if (value == classification)
+                    return $"{nameof(ClassificationTypeNames)}.{field.Name}";
+            }
+
+            return $"\"{classification}\"";
+        }
+
+        private static string GetKnownImageGuid(Guid guid)
+        {
+            foreach (var field in typeof(KnownImageIds).GetFields())
+            {
+                if (!field.IsStatic)
+                    continue;
+
+                var rawValue = field.GetValue(null);
+                var value = rawValue as Guid?;
```

Comment:
`Guid?`? Why the nullable? It seems like this changed slightly from the original VB.

Reply:
The `as` operator doesn't work with value types. You can either use a nullable value type, or you can use `is` followed by a cast. The original VB code used the latter, but `as` is cleaner in C#.

Reply:
Now simplified a bit (retained this concept though)